pkg/provider/azure_loadbalancer.go

```diff
@@ -1242,98 +1242,217 @@ func (az *Cloud) reconcileLoadBalancer(clusterName string, service *v1.Service,
 	}
 	lbName := *lb.Name
 	lbResourceGroup := az.getLoadBalancerResourceGroup()
+	lbBackendPoolID := az.getBackendPoolID(lbName, az.getLoadBalancerResourceGroup(), getBackendPoolName(clusterName, service))
```

Comment:
I had to do some pretty heavy refactoring of the `reconcileLoadBalancer` function (cyclo complexity was 76 before!) to get it under 30, please review extra closely.